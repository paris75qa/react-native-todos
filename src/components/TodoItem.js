import React, {PropTypes} from 'react';
import { TouchableHighlight, StyleSheet, Text } from 'react-native';

const styles = StyleSheet.create({
  text: {
    fontSize: 32,
  },
  complete: {
    textDecorationLine: "line-through",
  }
});
const TodoItem = (props) => {

    let completeStyle;
    if(props.complete) completeStyle = styles.complete;

    return (<TouchableHighlight onPress={props.onClick} ><Text style={[styles.text, completeStyle]} >{props.label}</Text></TouchableHighlight>);
}

TodoItem.propTypes = {
  label: PropTypes.string.isRequired,
  complete: PropTypes.bool,
  onClick: PropTypes.func,
}

export default TodoItem;
