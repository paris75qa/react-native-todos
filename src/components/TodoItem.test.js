import React from 'react';
import TodoItem from './TodoItem';
import renderer from 'react-test-renderer';
const label = "Item"

it('TodoItem renders correctly', () => {
  const tree = renderer.create(
    <TodoItem label="Item" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoItem renders correctly', () => {
  const tree = renderer.create(
    <TodoItem label={label} complete={true}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoItem label, complete = false', () => {
  const tree = renderer.create(
    <TodoItem label={label} complete={false}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoItem label, complete et onClick = vide', () => {
  const tree = renderer.create(
    <TodoItem label={label} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoItem onClick TEST', () => {
  const tree = renderer.create(
    <TodoItem label={label} onClick={()=>{}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
