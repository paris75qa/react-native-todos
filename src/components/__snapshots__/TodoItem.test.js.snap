// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem label, complete = false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        undefined,
      ]
    }
  >
    Item
  </Text>
</View>
`;

exports[`TodoItem label, complete et onClick = vide 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        undefined,
      ]
    }
  >
    Item
  </Text>
</View>
`;

exports[`TodoItem onClick TEST 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        undefined,
      ]
    }
  >
    Item
  </Text>
</View>
`;

exports[`TodoItem renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        undefined,
      ]
    }
  >
    Item
  </Text>
</View>
`;

exports[`TodoItem renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        Object {
          "textDecorationLine": "line-through",
        },
      ]
    }
  >
    Item
  </Text>
</View>
`;
